### PR TITLE
Qsfs autobuild

### PR DESCRIPTION
**Pull request by muhamadazmy**
Autobuild qsfs flist from the docker image from `quatntum-storage` this way newer version of qsfs can be easily integrated and used in the system

### CODE REVIEW

File: bins/bins-extra.sh [hunk 1]
```diff
@@ -144,6 +144,62 @@ download_file() {
...
+    # set default url
+    fileurl=$1
+
+    docker pull "${fileurl}"
+    docker save "${fileurl}" -o "${output}"
+}
+
+extract_docker_image() {
+    # you need to run `docker save <image> -o <file>.tar
+    # then run this script like `extrat-image.sh <file>.tar`
```
Comment (xmonader) [inline bins/bins-extra.sh]: extrat -> extract

File: bins/bins-extra.sh [hunk 2]
```diff
@@ -144,6 +144,62 @@ download_file() {
...
+    # it will create a directoyr `<file>.tar-root` which containers
```
Comment (xmonader) [inline bins/bins-extra.sh]: directoyr -> directory, which -> where

File: bins/packages/qsfs/qsfs.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+QSFS_VERSION="0.2.0-rc1"
```
Comment (xmonader) [inline bins/packages/qsfs/qsfs.sh]: v0.3.0 was released https://github.com/threefoldtech/0-stor_v2/releases/tag/v0.3.0 for 0-stor does that reflect on qsfs version?

File: bins/packages/qsfs/qsfs.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+install_qsfs() {
+    echo "[+] install qsfs"
+
+    mkdir -p "${ROOTDIR}"
+
+    extract_docker_image "${DISTDIR}/qsfs-${QSFS_VERSION}.tar" "${ROOTDIR}"
+}
+
+build_qsfs() {
+    pushd "${DISTDIR}"
```
Comment (xmonader) [inline bins/packages/qsfs/qsfs.sh:23]: variable should be DESTDIR instead of DISTDIR

File: bins/packages/qsfs/qsfs.sh (hunk 3 above)
Comment (muhamadazmy) [reply bins/packages/qsfs/qsfs.sh]: well, that's the version used by quatnum-storage https://github.com/threefoldtech/quantum-storage/blob/qsfs-docker/docker/Dockerfile#L4

File: bins/packages/qsfs/qsfs.sh (hunk 4 above)
Comment (muhamadazmy) [reply bins/packages/qsfs/qsfs.sh:23]: why? this variable is set by the base build script in the project and is used everywhere (in all packages) so changing that here means i need to change it everywhere 
also i think distdir is from (distribution dir) so it is correct

File: bins/bins-extra.sh (hunk 2 above)
Comment (muhamadazmy) [reply bins/bins-extra.sh]: I will re-write the entire description because when i started i had something else in mind

**Review by xmonader — APPROVED** [review]

File: bins/packages/qsfs/qsfs.sh (hunk 3 above)
Comment (muhamadazmy) [reply bins/packages/qsfs/qsfs.sh]: Updated, i had to rebuild qsfs docker to use latest version of zstor

**Pull request merged**